management/api-client.go

```diff
@@ -0,0 +1,109 @@
+package ziti
+
+import (
+	"crypto/tls"
+	"github.com/go-openapi/runtime"
+	openapiclient "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
+	"github.com/openziti/edge-api/rest_management_api_client"
+	"github.com/openziti/edge-api/rest_util"
+	"net"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"time"
+)
+
+type AuthenticatedApiClient struct {
+	*ApiClient
+	ZtSession     string
+	Authorization []string
+}
+
+func (a AuthenticatedApiClient) AuthenticateRequest(request runtime.ClientRequest, _ strfmt.Registry) error {
+	if err := request.SetHeaderParam("zt-session", a.ZtSession); err != nil {
+		return err
+	}
+
+	if err := request.SetHeaderParam("Authorization", a.Authorization...); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func NewAuthenticatedApiClient(apiSessionToken string, apiClient *ApiClient) *AuthenticatedApiClient {
+	authenticatedClient := &AuthenticatedApiClient{
+		ApiClient: apiClient,
+		ZtSession: apiSessionToken,
+	}
+
+	apiClient.Runtime.DefaultAuthentication = authenticatedClient
+
+	return authenticatedClient
```

Comment:
this is _our_ api client. we _know_ that it'll always need to be authenticated. adding the word "authenticated" is entirely redundant to me for that reason. I think "NewApiClient" is more succinct and jsut as descriptive

Reply:
Not all API requests are authenticated and thus not all clients need to be authenticated.